modules/core/src/shaderlib/project/project.glsl.js

```diff
@@ -55,20 +55,47 @@ const vec3 ZERO_64_LOW = vec3(0.0);
 const float EARTH_RADIUS = 6370972.0; // meters
 const float GLOBE_RADIUS = 256.0;
 
+// returns an adjustment factor for uCommonUnitsPerMeter
+float project_size() {
+  if (project_uProjectionMode == PROJECTION_MODE_WEB_MERCATOR &&
+    project_uCoordinateSystem == COORDINATE_SYSTEM_LNGLAT) {
+
+    // uCommonUnitsPerMeter in low-zoom Web Mercator is non-linear
+    // Adjust by 1 / cos(latitude)
+    // If geometry.position (vertex in common space) is populated, use it
+    // Otherwise use geometry.worldPosition (anchor in world space)
+    
+    if (geometry.position.w == 0.0) {
```

Comment:
Can this be precalculated so we don't need if statements (they tend to slow things down)?

Reply:
Not without breaking existing layers. 

This test will yield the same result for all vertices in a given layer, so there won't be any sync points.